src/main.rs

```diff
@@ -4,97 +4,135 @@ use std::collections::HashMap;
-        ),
-    ),
-    (
-        "test_rust",
-        ("Run all Rust tests", "sh ./automation/all_rust_tests.sh"),
-    ),
-    (
-        "verify_env",
-        (
-            "Verify your development environment",
+#[derive(Clone)]
+struct DialogItem {
+    name: &'static str,
+    nested: Option<Dialog>,
+    command: Option<&'static str>,
+    description: &'static str,
+}
+
+#[derive(Clone)]
+struct Dialog {
+    prompt: &'static str,
+    items: &'static [DialogItem],
+}
+
+const COMMANDS: &[DialogItem] = &[
+    DialogItem {
+        name: "build",
+        nested: None,
+        command: Some("cargo build"),
+        description: "Build Application Services",
+    },
+    DialogItem {
+        name: "test",
+        nested: None,
+        command: Some("sh ./automation/all_tests.sh"),
+        description: "Run all tests for a pull request",
+    },
+    DialogItem {
+        name: "test_rust",
+        command: Some("sh ./automation/all_rust_tests.sh"),
+        nested: None,
+        description: "Run all Rust tests",
+    },
+    DialogItem {
+        name: "verify_env",
+        description: "Verify your development environment",
+        command: Some(
             "sh ./libs/verify-desktop-environment.sh ;
-            sh ./libs/verify-android-environment.sh ;
-            sh ./libs/verify-ios-environment.sh",
-        ),
-    ),
-    (
-        "smoke_android",
-        (
-            "Smoke test Android Components",
-            "python3 ./automation/smoke-test-android-components.py",
-        ),
-    ),
-    (
-        "smoke_fenix",
-        (
-            "Smoke test Fenix Components",
-            "python3 ./automation/smoke-test-fenix.py",
+        sh ./libs/verify-android-environment.sh ;
+        sh ./libs/verify-ios-environment.sh",
         ),
-    ),
-    (
-        "smoke_ios",
-        (
-            "Smoke test Firefox iOS Components",
-            "python3 ./automation/smoke-test-firefox-ios.py",
-        ),
-    ),
-    (
-        "tag_minor",
-        (
-            "Tag a new release minor release",
-            "python3 ./automation/prepare-release.py minor",
-        ),
-    ),
-    (
-        "prepare_release",
-        (
-            "Tag a new release patch release",
-            "python3 ./automation/prepare-release.py patch",
-        ),
-    ),
-    (
-        "lint_bash",
-        (
-            "Lint bash script changes",
-            "sh ./automation/lint_bash_scripts.sh",
-        ),
-    ),
-    (
-        "cargo_update",
-        (
-            "Create a 'cargo update' PR",
-            "python3 ./automation/cargo-update-pr.py",
-        ),
-    ),
+        nested: None,
+    },
+    DialogItem {
+        name: "release",
+        description: "Prepare a release",
+        nested: Some(Dialog {
+            prompt: "What type of release would like to prepare?",
+            items: &[
+                DialogItem {
+                    name: "release-minor",
```

Comment:
would we be able to get the possible version number here?

Reply:
Alrighty, added them. I'm getting them from the `.buildconfig-android.yml` (is that the right place?), As long as the file is there and the `libraryVersion` is the first line and has with the`num.num.num` pattern then it works 😅

Reply:
Yes!! that's a good location to get that from

Reply:
Sweet! Added some error handling to make sure the tool doesn't crash if the file is not there (or the libraryVersion is not set) and instead just not print the versions in those cases (Also helpful testing since I don't need to be in the a-s directly all the time if I'm just changing UI stuff)